orchagent/dash/dashorch.cpp

```diff
@@ -0,0 +1,299 @@
+        {
+            return parseHandleSaiStatusFailure(handle_status);
+        }
+    }
+
+    appliance_entries_.erase(appliance_id);
+
+    SWSS_LOG_INFO("Removed vip and direction lookup entries for %s", appliance_id.c_str());
+    return true;
+}
+
+void DashOrch::doTaskApplianceTable(Consumer& consumer)
+{
+    SWSS_LOG_ENTER();
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
+        string appliance_id = kfvKey(t);
+        string op = kfvOp(t);
+
+        if (op == SET_COMMAND)
+        {
+            ApplianceEntry entry;
+
+            for (auto i : kfvFieldsValues(t))
+            {
+                if (fvField(i) == "sip")
+                {
+                    entry.sip = IpAddress(fvValue(i));
+                }
+                else if (fvField(i) == "vm_vni")
+                {
+                    entry.vm_vni = to_uint<uint32_t>(fvValue(i));
+                }
+            }
+            if (addApplianceEntry(appliance_id, entry))
+            {
+                it = consumer.m_toSync.erase(it);
+            }
+            else
+            {
+                it++;
+            }
+        }
+        else if (op == DEL_COMMAND)
+        {
+            if (removeApplianceEntry(appliance_id))
+            {
+                it = consumer.m_toSync.erase(it);
+            }
+            else
+            {
+                it++;
+            }
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Unknown operation %s", op.c_str());
+            it = consumer.m_toSync.erase(it);
+        }
+    }
+}
+
+bool DashOrch::addRoutingTypeEntry(const string& routing_type, const RoutingTypeEntry &entry)
+{
+    SWSS_LOG_ENTER();
+
+    if (routing_type_entries_.find(routing_type) != routing_type_entries_.end())
+    {
+        SWSS_LOG_INFO("Routing type entry already exists for %s", routing_type.c_str());
+        return true;
+    }
+
+    routing_type_entries_[routing_type] = entry;
+
+    SWSS_LOG_ERROR("Routing type entry added %s", routing_type.c_str());
+    return true;
+}
+
+bool DashOrch::removeRoutingTypeEntry(const string& routing_type)
+{
+    SWSS_LOG_ENTER();
+
+    if (routing_type_entries_.find(routing_type) == routing_type_entries_.end())
+    {
+        SWSS_LOG_INFO("Routing type entry does not exist for %s", routing_type.c_str());
+        return true;
+    }
+
+    routing_type_entries_.erase(routing_type);
+
+    SWSS_LOG_ERROR("Routing type entry removed for %s", routing_type.c_str());
+    return true;
+}
+
+void DashOrch::doTaskRoutingTypeTable(Consumer& consumer)
+{
+    SWSS_LOG_ENTER();
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
+        string routing_type = kfvKey(t);
+        string op = kfvOp(t);
+
+        if (op == SET_COMMAND)
+        {
+            RoutingTypeEntry entry;
```

Comment:
As captured in schema, this can come as a list of actions. Please handle

Reply:
Should we wait until we've finalized how multiple actions will be represented in the schema to change this?

Reply:
I will defer this change for now and handle it as a subsequent commit.